openlibrary/coverstore/disk.py

```diff
@@ -50,180 +49,6 @@ def exists(filename):
-        from web.utils import numify, denumify
-
-        #@@ this could be dangerous. If we clear the directory, it starts again from count 0.
-        #@@ Probably, this should be taken from database.
-        if self.next_warcfile is None:
-            files = [os.path.basename(f) for f in self.find(self.root) if f.endswith('.warc')]
-            if files:
-                files.sort()
-                self.next_warcfile = files[-1]
-            else:
-                self.next_warcfile = self.warcfile_prefix + '_0000_00.warc'
-
-        path = self.get_path(self.next_warcfile)
-        if os.path.exists(path) and self.filesize(path) >= self.maxsize:
-            count = int(numify(self.next_warcfile)) + 1
-            self.next_warcfile = self.warcfile_prefix + denumify("%06d" % count, "_XXXX_XX.warc")
-
-        return self.next_warcfile
-
-    def filesize(self, filename):
-        return os.stat(filename).st_size
-
-class ArchiveDisk(WARCDisk):
-    """Disk interface to internet archive storage.
-
-    There is a convention that is used to name files and items.
-    prefix_xxxx_yy.ext is saved in item named prefix_xxxx.
-    """
-    def make_warcfile(self, warcfilename):
-        path = self.get_path(warcfilename)
-        if os.path.exists(path):
-            return open(path)
-        else:
-            itemname = self.get_item_name(warcfilename)
-            url = self.item_url(itemname) + '/' + warcfilename
-            return warc.HTTPFile(url)
-
-    def read(self, filename):
-        # if the file is locally available then read it from there.
-        # else contact the server
-        try:
-            data = WARCDisk.read(self, filename)
-        except IOError:
-            data = None
-
-        if data:
-            return data
-
-        warcfilename, offset, size = filename.split(':')
-        offset = int(offset)
-        size = int(size)
-        f = self.make_warcfile(warcfilename)
-        f.seek(offset)
-        return f.read(size)
-
-    def create_file(self, filename):
-        itemname = self.get_item_name(filename)
-        url = self.item_url(itemname) + '/' + filename
-        return warc.HTTPFile(url)
-
-    def get_item_name(self, warcfilename):
-        # warc file file_xxxx_yy.warc is stored in item file_xxxx.
-        itemname, _ = warcfilename.rsplit('_', 1)
-        return itemname
-
-    def item_url(self, itemname):
-        """Returns http url to access files from the item specified by the itemname."""
-        from xml.dom import minidom
-        import urllib
-        base_url = 'http://www.archive.org/services/find_file.php?loconly=1&file='
-        try:
-            data= urllib.urlopen(base_url + itemname).read()
-            doc = minidom.parseString(data)
-            vals = ['http://' + e.getAttribute('host') + e.getAttribute('dir') for e in doc.getElementsByTagName('location')]
-            return vals and vals[0]
-        except Exception:
-            return None
 
 class LayeredDisk:
```

Comment:
It looks to me that you can delete `LayeredDisk` as well -- it seems to only have 1 hit in search, namely its definition.

https://github.com/internetarchive/openlibrary/search?q=layereddisk&unscoped_q=layereddisk

I haven't confirmed this, and I wouldn't block this PR on that, just a FYI.